### PR TITLE
Simple Fallback Node Implementation

### PR DESCRIPTION
**Pull request by calvertdw**
image

### CODE REVIEW

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeExecutor.java [hunk 1]
```diff
@@ -72,10 +72,10 @@ public void destroy()
 
    public void setRootNode(BehaviorTreeNodeLayer<?, ?, ?, ?> rootNode)
    {
-      this.rootNode = (BehaviorTreeNodeExecutor<?, ?>) rootNode;
+      this.rootNode = (BehaviorTreeRootNodeExecutor) rootNode;
    }
 
-   public BehaviorTreeNodeExecutor<?, ?> getRootNode()
+   public BehaviorTreeRootNodeExecutor getRootNode()
```
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeExecutor.java:78]: Should have been done long ago

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java
Comment (calvertdw) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java:1]: The fallback logic needs to happen alongside the sequence logic. It's very coupled. The main logic is in here.

**Review by LuigiPenco93 — COMMENTED** [review]
can we rename the fallback - failure/ success in the UI to defaults as "try actions" and "fallback actions", to be consistent with the  nomenclature you have in the code.

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java [hunk 2]
```diff
@@ -120,16 +176,18 @@ else if (state.getActionChildren().get(i).calculateExecuteAfterActionIndex() < e
...
-            state.getLogger().error("An action failed. Disabling automatic execution.");
+            state.getLogger().error("An action failed. Disabling automatic execution.\n   Failed: %s".formatted(failedActionsWithoutFallback));
             state.setAutomaticExecution(false);
          }
          else
          {
             while (shouldExecuteNextAction())
             {
-               state.getLogger().info("Automatically executing action: {}", actionChildren.get(state.getExecutionNextIndex()).getClass().getSimpleName());
+               var nextAction = actionChildren.get(state.getExecutionNextIndex());
```
Comment (TomaszTB) [inline ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java:188]: Hmm `var`? That's cool. We don't have any style guidelines against it, do we?

**Review by TomaszTB — APPROVED** [review]
Looks good to me

File: ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java (hunk 2 above)
Comment (calvertdw) [reply ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java:188]: I think it's totally fine in places like this where the type is clunky and not important for understanding what's going on

**Review by LuigiPenco93 — APPROVED** [review]

**Review by ds58 — APPROVED** [review]

**Pull request merged**